backvendor/workingtree.go

```diff
@@ -34,16 +33,68 @@ import (
 	"golang.org/x/tools/go/vcs"
 )
 
-// A WorkingTree is a local checkout of Go source code, and
-// information about the version control system it came from.
-type WorkingTree struct {
+var execCommand = exec.Command
+
+// Describable is the interface which capture the methods required for
+// creating a pseudo-version from a revision.
+type Describable interface {
+	// ReachableTag returns the most recent reachable tag,
+	// preferring semver tags. It returns ErrorVersionNotFound if
+	// no suitable tag is found.
+	ReachableTag(rev string) (string, error)
+
+	// TimeFromRevision returns the commit timestamp from the
+	// revision rev.
+	TimeFromRevision(rev string) (time.Time, error)
+}
+
+// A WorkingTree is a local checkout of Go source code, and methods to
+// interact with the version control system it came from.
+type WorkingTree interface {
+	io.Closer
+
+	// Should be something that supports creating pseudo-versions.
+	Describable
+
+	// TagSync syncs the repo to the named tag.
+	TagSync(tag string) error
+
+	// VersionTags returns the semantic version tags.
+	VersionTags() ([]string, error)
+
+	// Revisions returns all revisions starting from the newest.
```

Comment:
In a particular order?  If 'starting' wasn't mentioned then I'd assume no order. With 'starting' it's partially ordered. :-)

Reply:
Updated.